### PR TITLE
Implement VMIN and VMAX Evaluator

### PR DESCRIPTION
**Pull request by r30shah**
Signed-off-by: Rahil Shah <rahil@ca.ibm.com>

### CODE REVIEW

**Comment by r30shah** [comment]
@joransiu  Can I please get your review on this one?

**Review by joransiu — APPROVED** [review]
LGTM.  One minor wording nit.

File: compiler/z/codegen/OMRTreeEvaluator.cpp [hunk 1]
```diff
@@ -1174,13 +1174,27 @@ OMR::Z::TreeEvaluator::vsqrtEvaluator(TR::Node *node, TR::CodeGenerator *cg)
...
 OMR::Z::TreeEvaluator::vminEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   return TR::TreeEvaluator::unImpOpEvaluator(node, cg);
+   TR_ASSERT_FATAL_WITH_NODE(node, node->getDataType().getVectorLength() == TR::VectorLength128,
+                   "Only 128-bit vectors are supported %s", node->getDataType().toString());
+   TR::DataType dt = node->getDataType().getVectorElementType();
+   TR_ASSERT_FATAL_WITH_NODE(node, cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_S390_VECTOR_FACILITY_ENHANCEMENT_1) ||
+                                       !(dt == TR::Double ||
+                                          dt == TR::Float),
+                                    "VFMIN is only supported on Z14 onwards.");
```
[the remark this answers is not in the document]
Comment (r30shah) [reply compiler/z/codegen/OMRTreeEvaluator.cpp]: Fixed in https://github.com/eclipse/omr/compare/c06883a50773377be88a23f5360647c5073ce626..c65f22758a48c86e44cbd02e35df6347e737ab96

**Comment by r30shah** [comment]
Jenkins build zos,zlinux

**Comment by r30shah** [comment]
Failures seen on macOS and this changes are not related. Rebased the changes. Launching build again.

**Comment by r30shah** [comment]
Jenkins build zos,zlinux

**Comment by r30shah** [comment]
Jenkins build zos,zlinux

**Comment by r30shah** [comment]
@0xdaryl  Can I request you to review and merge this PR? Changes are Z specific and builds on Z passes.

File: compiler/z/codegen/OMRTreeEvaluator.cpp [hunk 2]
```diff
@@ -1174,13 +1174,27 @@ OMR::Z::TreeEvaluator::vsqrtEvaluator(TR::Node *node, TR::CodeGenerator *cg)
...
    {
-   return TR::TreeEvaluator::unImpOpEvaluator(node, cg);
+   TR_ASSERT_FATAL_WITH_NODE(node, node->getDataType().getVectorLength() == TR::VectorLength128,
+                   "Only 128-bit vectors are supported %s", node->getDataType().toString());
+   TR::DataType dt = node->getDataType().getVectorElementType();
+   TR_ASSERT_FATAL_WITH_NODE(node, cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_S390_VECTOR_FACILITY_ENHANCEMENT_1) ||
+                                       !(dt == TR::Double ||
+                                          dt == TR::Float),
+                                    "VFMAX is only supported on z14 onwards.");
+   return TR::TreeEvaluator::inlineVectorBinaryOp(node, cg, (dt == TR::Double || dt == TR::Float) ? TR::InstOpCode::VFMAX : TR::InstOpCode::VMX);
```
Comment (0xdaryl) [inline compiler/z/codegen/OMRTreeEvaluator.cpp:1197]: Can `vminEvaluator` and `vmaxEvaluator` call a common function to save some code footprint and avoid duplication?  Their implementations are identical except for the ultimate choice of InstOpCode.

Comment (r30shah) [reply compiler/z/codegen/OMRTreeEvaluator.cpp:1176]: Yes I have refactored the evaluators to actually do a safe check in the common function which are used to generate instructions in https://github.com/eclipse/omr/pull/6618/commits/92e99ed96c9294f949b12983899fc70ba6d6b5df

I also noticed some other vector evaluators on Z which are doing the same thing and can get benefit with some refactoring, so opened up OMR issue to come back to it later (https://github.com/eclipse/omr/issues/6654)

**Comment by r30shah** [comment]
Jenkins build zos,zlinux

**Comment by r30shah** [comment]
MacOS failure in the build (https://dev.azure.com/eclipse-omr/omr/_build/results?buildId=6220&view=logs&j=aa2fd561-a338-5537-45ee-49ae22aea958&t=20ff6471-47a7-5631-6842-540dac3e3fb4&l=1888) seems to be known issue [1], and changes in this PR are Z code-gen specific so I do not think it can  cause any failures on x86-MacOS. 

@0xdaryl  I have tried corporate your suggestion in https://github.com/eclipse/omr/pull/6618#discussion_r944463263, can we merge this one if you are OK with the changes? 


[1]. https://github.com/eclipse/omr/issues/6556

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**